indexbuilder.go

```diff
@@ -191,11 +199,12 @@ func (b *IndexBuilder) ContentSize() uint32 {
 
 // NewIndexBuilder creates a fresh IndexBuilder. The passed in
 // Repository contains repo metadata, and may be set to nil.
-func NewIndexBuilder(r *Repository) (*IndexBuilder, error) {
+func NewIndexBuilder(r *Repository, iopts *IndexOptions) (*IndexBuilder, error) {
```

Comment:
having to pass in indexoptions here almost makes me feel like largefiles should be part of the repo. But I'll leave the decision up to you.